src/xtopcom/xrpc/xgetblock/get_block.cpp

```diff
@@ -1514,6 +1565,45 @@ xJson::Value get_block_handle::get_block_json(xblock_t * bp, const std::string &
     return root;
 }
 
+xJson::Value get_block_handle::get_blocks_json(xblock_t * bp, const std::string & rpc_version) {
+    xJson::Value root;
+    if (bp == nullptr) {
+        return root;
+    }
+
+    if (bp->is_genesis_block() && bp->get_block_class() == base::enum_xvblock_class_nil && false == bp->check_block_flag(base::enum_xvblock_flag_stored)) {
+        // genesis empty non-stored block, should not return
+        return root;
+    }
+
+    // load input for raw tx get
+    if (false == base::xvchain_t::instance().get_xblockstore()->load_block_input(base::xvaccount_t(bp->get_account()), bp, metrics::blockstore_access_from_rpc_get_block_json)) {
+        xassert(false);  // db block should always load input success
+        return root;
+    }
+
+    set_shared_info(root, bp);
+
+    xJson::Value header;
+    set_header_info(header, bp);
+    root["header"] = header;
+
+    xJson::Value body;
+    set_body_info(body, bp, rpc_version);
+
+    root["body"] = body;
+    if (bp->check_block_flag(base::enum_xvblock_flag_committed)) {
```

Comment:
duplicate code, why not update the implementation of `get_block_json` directly?

Reply:
next version fix it